ParkingZoneApp/Areas/Admin/Controllers/ParkingZonesController.cs

```diff
@@ -1,36 +1,36 @@
 ﻿using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
-using ParkingZoneApp.Data;
 using ParkingZoneApp.Models;
+using ParkingZoneApp.Repository.Interfaces;
 
 namespace ParkingZoneApp.Areas.Admin
 {
     [Area("Admin")]
     [Authorize]
     public class ParkingZonesController : Controller
     {
-        private readonly ApplicationDbContext _context;
 
-        public ParkingZonesController(ApplicationDbContext context)
+        private readonly IRepository<ParkingZone> genericRepository;
```

Comment:
implement from IParkingZoneRepository why did you added it? in order to use here

Reply:
I see